03-ServicePractice/SetValueApp/app/src/main/java/com/example/setvalueapp/MainActivity.java

```diff
@@ -0,0 +1,119 @@
+package com.example.setvalueapp;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.content.ComponentName;
+import android.content.Context;
+import android.content.Intent;
+import android.content.ServiceConnection;
+import android.os.Bundle;
+import android.os.IBinder;
+import android.util.Log;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import com.example.valueservice.IValueService;
+
+public class MainActivity extends AppCompatActivity {
+    static final String LOG_TAG = "SetValueApp";
+
+    private IValueService mService;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        Log.d(LOG_TAG, "-> onCreate() MainActivity");
+        setContentView(R.layout.activity_main);
+
+        Button buttonStartService = findViewById(R.id.StartServiceButton);
+        buttonStartService.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.d(LOG_TAG, "-> onClick(StartService)");
+                if (mService!=null) return;
+                TextView editTextSetValue = findViewById(R.id.ResultTextView);
+                if (bindSrv()) {
+                    Log.d(LOG_TAG, "-> onClick(StartService): bindSrv() has returned TRUE!");
```

Comment:
It does not mean that service is bonded.
From developer.android.com:

> true if the system is in the process of bringing up a service that your client has permission to bind to; false if the system couldn't find the service or if your client doesn't have permission to bind to it. If this value is true, you should later call unbindService(ServiceConnection) to release the connection.

Service is bonded successful when ServiceConnection.onServiceConnected is called.

Reply:
Fixed. Please reveiw.

Reply:
Also fixed in the GetValueApp.